clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -966,12 +966,12 @@ public void subscribe(Collection<String> topics) {
      */
     @Override
     public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
-        acquireAndEnsureOpen();
         try {
             if (pattern == null)
                 throw new IllegalArgumentException("Topic pattern to subscribe to cannot be null");
 
             throwIfNoAssignorsConfigured();
+            acquireAndEnsureOpen();
```

Comment:
You need to fix these other methods also. If the exception is raised, the call to `release` will lead to invalid state.

Reply:
@hachikuji Sure will fix the rest. Thanks!